lib/min_heap.rb

```diff
@@ -52,20 +61,48 @@ def empty?
 
   private
 
+  def find_parent(index)
+    parent_index = (index - 1)/2
+    return parent_index
+  end
+
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
   # Time complexity: ?
   # Space complexity: ?
   def heap_up(index)
-    
+    puts to_s
+    parent_index = find_parent(index)
+    if @store[parent_index].key <= @store[index].key || index == 0
+      return
+    end 
+    if @store[parent_index].key > @store[index].key
+      # puts  @store[index].value
+      swap(parent_index, index)
+      index = parent_index
+      heap_up(index)
+    end
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
-    raise NotImplementedError, "Method not implemented yet..."
+    puts 'heap down'
+    puts to_s
+    child_index = 2*index + 1
```

Comment:
You are only finding the left child here.  You are **not** finding the right child.